src/zenml/utils/source_utils.py

```diff
@@ -582,33 +590,61 @@ def _try_to_load_notebook_source(source: NotebookSource) -> Any:
 
     Raises:
         RuntimeError: If the source can't be loaded.
+        FileNotFoundError: If the file containing the notebook cell code can't
+            be found.
 
     Returns:
         The loaded object.
     """
-    if not source.code_path or not source.replacement_module:
+    if not source.replacement_module:
         raise RuntimeError(
             f"Failed to load {source.import_path}. This object was defined in "
             "a notebook and you're trying to load it outside of a notebook. "
-            "This is currently only enabled for ZenML steps."
+            "This is currently only enabled for ZenML steps and materializers. "
+            "To enable this for your custom classes or functions, use the "
+            "`zenml.utils.notebook_utils.enable_notebook_code_extraction` "
+            "decorator."
         )
 
     extract_dir = _get_shared_temp_dir()
-    file_path = os.path.join(extract_dir, f"{source.replacement_module}.py")
+    file_name = f"{source.replacement_module}.py"
+    file_path = os.path.join(extract_dir, file_name)
 
     if not os.path.exists(file_path):
+        from zenml.client import Client
         from zenml.utils import code_utils
 
+        artifact_store = Client().active_stack.artifact_store
+
+        if (
+            source.artifact_store_id
+            and source.artifact_store_id != artifact_store.id
+        ):
+            raise RuntimeError(
+                "Notebook cell code not stored in active artifact store."
+            )
+
         logger.info(
-            "Downloading notebook cell content from `%s` to load `%s`.",
-            source.code_path,
+            "Downloading notebook cell content to load `%s`.",
             source.import_path,
         )
 
-        code_utils.download_and_extract_code(
-            code_path=source.code_path, extract_dir=extract_dir
-        )
+        try:
+            code_utils.download_notebook_code(
+                artifact_store=artifact_store,
+                file_name=file_name,
+                download_path=file_path,
+            )
+        except FileNotFoundError:
+            if not source.artifact_store_id:
+                raise FileNotFoundError(
+                    "Unable to find notebook code file. This might be because "
```

Comment:
With the error message above, this one is now a bit misleading as it will fail if the file is stored in a different artifact store.